pkg/controller/argocd/custommapper.go

```diff
@@ -96,3 +97,30 @@ func (r *ReconcileArgoCD) tlsSecretMapper(o handler.MapObject) []reconcile.Reque
 
 	return result
 }
+
+func (r *ReconcileArgoCD) namespaceResourceMapper(o handler.MapObject) []reconcile.Request {
+	var result = []reconcile.Request{}
+
+	labels := o.Meta.GetLabels()
+	if v, ok := labels[common.ArgoCDManagedNamespaceLabel]; ok {
+		argocds := &argoprojv1alpha1.ArgoCDList{}
```

Comment:
According to the design it is was decided to use the <source namespace> as the value for the label to identify the ArgoCD instance. 
But it does seem like a better idea to use an identifier `namespace/name`. That way i wouldn't need to list ArgoCD instances, i can directly query for the particular instance. @jannfis @sbose78 thoughts?

Reply:
That sounds tempting, but we need to take care not to hit the 63 characters limit for labels here. Names of namespaces can become pretty long already (they max out at 63 chars as well). So if someone has a long namespace name, this may not work.

Reply:
That one extra API Call should be fine :) the lesser the user does, the more accurate he will be.